### ISSUE TITLE
`Cache` missing `Debug` impl

### ISSUE THREAD

**Issue by lilyball** [moka-rs/moka#132]
The `Cache` types are all missing `Debug` impls, which is annoying. I'm not sure what offhand such an impl should show, but perhaps some basic stats about the cache if that's easy to gather.

**Comment by tatsuya6502** [comment]
Thank you for reporting this issue.

> I'm not sure what offhand such an impl should show, but perhaps some basic stats about the cache if that's easy to gather.

I started to work on this as #134 and made the `Debug` impl to show some basic stats. But while I was writing an example for the doc, I realized this will confuse users as these stats are eventual consistent (will show stale numbers).

```rust
use moka::sync::Cache;
​
let cache = Cache::new(10);
​
// Insert entries.
cache.insert('d', "Dutch");
cache.insert('n', "Netherland Dwarf");
cache.insert('l', "Lop Eared");
​
// Ensure an entry exists.
assert_eq!(cache.get(&'d'), Some("Dutch"));
​
// Print a debug info. Note that this may print stale numbers.
// (e.g. `entry_count` can be `0` instead of `3`)
println!("{:?}", cache);
// -> Cache { max_capacity: Some(10), entry_count: 0, weighted_size: 0 }
​
// You can mitigate this by performing a `sync` first.
// Bring `ConcurrentCacheExt` trait to the scope so we can use `sync` method.
use moka::sync::ConcurrentCacheExt;
// Call `sync` to run pending internal tasks.
cache.sync();

// Now it will print the actual numbers.
println!("{:?}", cache);
// -> Cache { max_capacity: Some(10), entry_count: 3, weighted_size: 3 }
```

These are kept eventual consistent to improve performance ([doc](https://docs.rs/moka/0.8.4/moka/#implementation-details)).

So I think I will change it to show key-value entries like the `Debug` impl of `std::collections::HashMap` does. It will be something like:

```rust
use moka::sync::Cache;
​
let cache = Cache::new(10);
​
// Insert entries.
cache.insert('d', "Dutch");
cache.insert('n', "Netherland Dwarf");
cache.insert('l', "Lop Eared");
​
// Print a debug info.
println!("{:?}", cache);
// -> {'l': "Lop Eared", 'd': "Dutch", 'n': "Netherland Dwarf"}
// Note that entries are returned in arbitrary order.
```

To show key-value entries, both `K` and `V` must implement `Debug`. I hope you are okay with this.

**Comment by lilyball** [comment]
If it's easy enough to show those and won't be too expensive then that sounds fine to me. It just bothered me that my type that has a cache couldn't show anything at all about it.

It looks like #134 is adding methods to query for these stats, which I appreciate as it means I have the option of showing those after all if it suits my usage better.

**Comment by tatsuya6502** [comment]
Hi @lilyball — I have published [Moka v0.8.5](https://crates.io/crates/moka/0.8.5) to crates.io containing this and some other changes. Thanks!

- [CHANGELOG Version v0.8.5](https://github.com/moka-rs/moka/blob/master/CHANGELOG.md#version-085)
